### PR TITLE
[RFC] tests: use &shellcmdflag instead of hardcoding -c

### PR DESCRIPTION
**Pull request by sethjackson**
Cherry picked from #810. From equalsraf@a8e1841.

### CODE REVIEW

File: test/functional/job/job_spec.lua [hunk 1]
```diff
@@ -228,10 +228,10 @@ describe('jobs', function()
...
       source([[
       call rpcnotify(g:channel, 'wait', jobwait([
-      \  jobstart([&sh, '-c', 'sleep 0.10; exit 4']),
-      \  jobstart([&sh, '-c', 'sleep 0.110; exit 5']),
-      \  jobstart([&sh, '-c', 'sleep 0.210; exit 6']),
-      \  jobstart([&sh, '-c', 'sleep 0.310; exit 7'])
+      \  jobstart([&sh, &shellcmdflag, 'sleep 0.10; exit 4']),
+      \  jobstart([&sh, &shellcmdflag, 'sleep 0.110; exit 5']),
+      \  jobstart([&sh, &shellcmdflag, 'sleep 0.210; exit 6']),
+      \  jobstart([&sh, &shellcmdflag, 'sleep 0.310; exit 7'])
```
Comment (justinmk) [inline test/functional/job/job_spec.lua]: sleep isn't typically available on Windows 7. The old [`ping` trick](http://ss64.com/nt/sleep.html)

    ping -n 4 -w 110 127.0.0.1>nul

even with the `-w` option of ping it seems not very precise. We can ship `sleep.exe` with the Windows build, but even if we do that we should skip these tests if `sleep` cannot be found, so that running the tests on Windows is not unpleasant for new contributors. 

```c
if eval("executable('sleep')") == 0 then
  pending('missing "sleep" command', function() end)
else
  it('returns a list of status codes', function()
  ...
```

reference: https://github.com/neovim/neovim/wiki/Unit-tests

Comment (ZyX-I) [inline test/functional/job/job_spec.lua]: Actually you should have here `jobstart(split(&shell) + split(&shellcmdflag) + ['sleep 0.10; exit 4'])`. Documentation that says that `jobstart('{cmd}')` has something to do with

    			:call jobstart([&shell, &shellcmdflag, '{cmd}'])

is incorrect. My variant is neither because &shell and &shellcmdflag are split with quotes taken into account (which I find rather useless because I have not found where it is unquoted and additionally it does not take backslash into account), but this is closer.

Comment (justinmk) [inline test/functional/job/job_spec.lua]: That change should be in a different commit, if it is added to this PR.

Comment (ZyX-I) [inline test/functional/job/job_spec.lua]: Also if you are going to take cmd.exe into account then these tests should be rewritten because it separates commands using `&` and not `;`.

Comment (justinmk) [inline test/functional/job/job_spec.lua]: In that case for these tests we should just use powershell. It also has `sleep`.

Comment (equalsraf) [inline test/functional/job/job_spec.lua]: > Documentation that says that jobstart('{cmd}') has something to do with

AFAIK jobstart('cmd') is using `shell_build_argv` that calls - p_sh p_shcf cmd (and splits). Could tests that are written as `jobstart([&sh, &shellcmdflag, ...` be rewritten as jobstart('...')?

Comment (justinmk) [inline test/functional/job/job_spec.lua]: Yeah they could, since it is orthogonal to the purpose of these tests.

**Comment by justinmk** [comment]
@sethjackson suggestions:

- in `before_each`: `:set shell=powershell shellcmdflag=-c shellquote=\" shellxquote=`
- write a local helper function `sleep_exit(ms, exitcode)` that returns string `"sleep 0.310; exit 7"` for non-Windows and `"sleep -milliseconds 310; exit 7"` for Windows.
